06-building-a-program-start-to-finish/solving-the-problem/output/inverted-index.js

```diff
@@ -0,0 +1,127 @@
+/* eslint-disable no-undef */
+
+/**
+ * inverted index class
+ * @class
+**/
+class InvertedIndex { //  eslint-disable-line
+  /**
+  * class constructor
+  * @constructor
+  **/
+  constructor() {
+    this.fileIndices = {};
+    this.searchIndices = {};
+  }
+  /**
+   * Set Index - Sets the indices of all indexed files
+   * @param {String} filename - Name of the indexed file
+   * @param {Object} indices - Indices of the file
+   * @return {object} Indexed file name and it's indices'
+  **/
+  setIndex(filename, indices) {
+    this.fileIndices[filename] = indices;
+  }
+  /**
+   * Validate File
+   * It checks if a json file is a json array of json objects
+   * @param {Object} file is an array of json objects
+   * @return {Boolean} True if a json file is valid and False otherwise
+  **/
+  validateFile(file) { // eslint-disable-line
+    if (typeof file !== 'object' || file.length === 0 || !Array.isArray(file)) {
+      return false;
+    }
+    for (let i = 0; i < file.length; i += 1) {
+      const item = file[i];
+      if (!(item.hasOwnProperty('title') && item.hasOwnProperty('text'))) { // eslint-disable-line
+        return false;
+      }
+    }
+    return true;
+  }
+  /**
+   * Tokenize
+   * It splits sentence into an array of refined words
+   * @param {String} text - string of texts
+   * @return {Array} An array of refined splitted texts
+  **/
+  tokenize(text) {  //eslint-disable-line
+    const remove = /[^'^\w\s]/g;
+    return text.replace(remove, ' ').toLowerCase().split(' ')
+    .sort()
+    .filter(item => Boolean(item));
+  }
+  createIndex(fileName, fileContent) {  //eslint-disable-line
+    const indices = {};
+    if (this.validateFile(fileContent)) {
+      fileContent.forEach((doc, docIndex) => {
+        const newString = `${doc.title} ${doc.text}`;
```

Comment:
Using data types in variable names is not allowed e.g. `newString`, `tokenArray`